### PR TITLE
Migrate to Hooks

### PR DESCRIPTION
**Pull request by mauriciord**
# Summary

### motivation

[migrate to hooks by Sibelius](https://github.com/react-native-community/react-native-camera/issues/2199)

* Get experience with hooks
* Get experience contributing to open source
* Learn in public

## Test Plan

Clone this example repository and see (... in progress)

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅       |
| Android |    ✅       |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [x] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

## TODO
- Migrate the file to TS (???)
- A way to access `takePictureAsync`
- Add in README
- Mention on CHANGELOG
- Insert my example repo on this PR description
- Fix CI checks

### CODE REVIEW

File: src/RNCamera.js [hunk 1]
```diff
@@ -342,159 +342,259 @@ const mapValues = (input, mapper) => {
...
-    useCamera2Api: PropTypes.bool,
-    playSoundOnCapture: PropTypes.bool,
-    videoStabilizationMode: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    pictureSize: PropTypes.string,
-    mirrorVideo: PropTypes.bool,
-    rectOfInterest: PropTypes.any,
-    defaultVideoQuality: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-  };
+export default function Camera(props) {
+  let _cameraRef: ?Object;
```
Comment (sibelius) [inline src/RNCamera.js]: useRef?

File: src/RNCamera.js [hunk 2]
```diff
@@ -342,159 +342,259 @@ const mapValues = (input, mapper) => {
...
+  let _cameraHandle: ?number;
+  let _lastEvents: { [string]: string };
+  let _lastEventsTimes: { [string]: Date };
+  let _isMounted: boolean;
```
Comment (sibelius) [inline src/RNCamera.js]: useRef

File: src/RNCamera.js [hunk 3]
```diff
@@ -342,159 +342,259 @@ const mapValues = (input, mapper) => {
...
+
+  const [isAuthorized, setIsAuthorized] = useState(false);
```
Comment (sibelius) [inline src/RNCamera.js]: useState<boolean>

**Comment by fabriziobertoglio1987** [comment]
to solve the following [merge conflict][1]: 

[1]: https://github.com/AgileVentures/MetPlus_PETS/wiki/Resolving-Pull-Request-merge-conflicts

you need to delete  `static propTypes = {}` from the `master` branch, as you included them already at the bottom of the file as `Camera.propTypes = { }`

```javascript
<<<<<<< feat/hooksample
export default function Camera(props) {
  let _cameraRef = useRef((ref) => setReference(ref));
  let _cameraHandle: ?number;
  let _lastEvents: { [string]: string };
  let _lastEventsTimes: { [string]: Date };
  let _isMounted: boolean;

  const [isAuthorized, setIsAuthorized] = useState<boolean>(false);
  const [isAuthorizationChecked, setIsAuthorizationChecked] = useState<boolean>(false);
  const [recordAudioPermissionStatus, setRecordAudioPermissionStatus] = useState(
    RecordAudioPermissionStatusEnum.PENDING_AUTHORIZATION,
  );

  const convertProp = useMemo((value: *, key: string) => {
    if (typeof value === 'string' && Camera.ConversionTables[key]) {
      return Camera.ConversionTables[key][value];
    }

    return value;
  }, []);
  const convertNativeProps = useCallback(
    (nativeProps: PropsType) => () => {
      const { children, ...props } = nativeProps;
      const newProps = mapValues(props, convertProp);

      if (props.onBarCodeRead) {
        newProps.barCodeScannerEnabled = true;
      }
======= DELETE FROM HERE
  static propTypes = {
    ...ViewPropTypes,
    zoom: PropTypes.number,
    maxZoom: PropTypes.number,
    ratio: PropTypes.string,
    focusDepth: PropTypes.number,
    onMountError: PropTypes.func,
    onCameraReady: PropTypes.func,
    onAudioInterrupted: PropTypes.func,
    onAudioConnected: PropTypes.func,
    onStatusChange: PropTypes.func,
    onBarCodeRead: PropTypes.func,
    onPictureTaken: PropTypes.func,
    onPictureSaved: PropTypes.func,
    onRecordingStart: PropTypes.func,
    onRecordingEnd: PropTypes.func,
    onGoogleVisionBarcodesDetected: PropTypes.func,
    onFacesDetected: PropTypes.func,
    onTextRecognized: PropTypes.func,
    onSubjectAreaChanged: PropTypes.func,
    trackingEnabled: PropTypes.bool,
    faceDetectionMode: PropTypes.number,
    faceDetectionLandmarks: PropTypes.number,
    faceDetectionClassifications: PropTypes.number,
    barCodeTypes: PropTypes.arrayOf(PropTypes.string),
    googleVisionBarcodeType: PropTypes.number,
    googleVisionBarcodeMode: PropTypes.number,
    type: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    cameraId: PropTypes.string,
    flashMode: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    exposure: PropTypes.number,
    whiteBalance: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    autoFocus: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
    autoFocusPointOfInterest: PropTypes.shape({ x: PropTypes.number, y: PropTypes.number }),
    permissionDialogTitle: PropTypes.string,
    permissionDialogMessage: PropTypes.string,
    androidCameraPermissionOptions: Rationale,
    androidRecordAudioPermissionOptions: Rationale,
    notAuthorizedView: PropTypes.element,
    pendingAuthorizationView: PropTypes.element,
    captureAudio: PropTypes.bool,
    keepAudioSession: PropTypes.bool,
    useCamera2Api: PropTypes.bool,
    playSoundOnCapture: PropTypes.bool,
    videoStabilizationMode: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    pictureSize: PropTypes.string,
    mirrorVideo: PropTypes.bool,
    rectOfInterest: PropTypes.any,
    defaultVideoQuality: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  };
>>>>>>> master TO HERE
```

**Pull request closed without merging**